src/machine.rs

```diff
@@ -151,6 +100,23 @@ impl<'m> Machine<'m> {
 
         // TODO: Handle fifos. See https://github.com/firecracker-microvm/firecracker-go-sdk/blob/f0a967ef386caec37f6533dce5797038edf8c226/jailer.go#L435
 
+        // FIXME: Assuming jailer for now.
+        let jailer = config.jailer_cfg.as_mut().expect("no jailer config");
+        let (daemonize_arg, stdin, stdout, stderr) = match &mut jailer.mode {
+            JailerMode::Daemon => (
+                Some("--daemonize"),
+                Stdio::null(),
+                Stdio::null(),
+                Stdio::null(),
+            ),
+            JailerMode::Attached(stdio) => (
+                None,
+                stdio.stdin.take().unwrap_or_else(Stdio::inherit),
+                stdio.stdout.take().unwrap_or_else(Stdio::inherit),
+                stdio.stderr.take().unwrap_or_else(Stdio::inherit),
+            ),
+        };
+
```

Comment:
Was this block of code moved here for a reason? :thinking:

Reply:
yeah, just to move it closer to where the vars are used

Reply:
Makes sense but to make reviewer's task easier, it's best to split such refactoring also in a separate commit, where possible. Otherwise, reviewer has to compare each line to see if anything actually changed. Now the "reviewer" knows so no need to change, just for future ref.